### PR TITLE
[hpx] Two new variants for upcoming v1.5, stable and master

### PR DESCRIPTION
**Pull request by teonnik**
- Remove cxxstd=17 for stable as building with cxxstd=14 is also possible.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hpx/package.py [hunk 1]
```diff
@@ -100,6 +101,10 @@ class Hpx(CMakePackage, CudaPackage):
 
     # MPI
     depends_on('mpi', when='networking=mpi')
+    depends_on('mpi', when='+async_mpi')
```
Comment (albestro) [inline var/spack/repos/builtin/packages/hpx/package.py:104]: Is there any relationship/constraint between `async_mpi` and `networking=mpi`?

Comment (teonnik) [reply var/spack/repos/builtin/packages/hpx/package.py:104]: Yes, there is, `networking=mpi` implies `async_mpi=True`. I am not sure how to describe the relationship in spack though. Ideally, I would be able to set the default value of `async_mpi` based on whether `networking=mpi`.

File: var/spack/repos/builtin/packages/cxxopts/package.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cxxopts/package.py]: Can you remove this spurious change?

File: var/spack/repos/builtin/packages/hpx/package.py (hunk 1 above)
Comment (albestro) [reply var/spack/repos/builtin/packages/hpx/package.py:104]: If it is not a constraint, so any combination of `networking` and `async_mpi` is valid, I think it is perfect like this. 👍

**Review by alalazo — DISMISSED** [review]
A minor request

**Review by albestro — APPROVED** [review]

**Comment by alalazo** [comment]
@teonnik Can you rebase and push to retrigger tests?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**